pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,138 @@ def wrapper(*args, **kwargs):
         )
 
     return wrapper
+
+
+@functools.lru_cache
+def _get_cond_qfunc_prim():
+    """Get the cond primitive for quantum functions."""
+
+    # JAX should be installed if capture is enabled
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = jax.core.Primitive("cond")
+    cond_prim.multiple_results = True
+
+    @cond_prim.def_impl
+    def _(condition, elifs_conditions, *args, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        def run_jaxpr(jaxpr, *args):
+            return jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
+
+        def true_branch(args):
+            return run_jaxpr(jaxpr_true, *args)
+
+        def elif_branch(args, elifs_conditions, jaxpr_elifs):
+            if not jaxpr_elifs:
+                return None
+            pred = elifs_conditions[0]
+            rest_preds = elifs_conditions[1:]
+            jaxpr_elif = jaxpr_elifs[0]
+            rest_jaxpr_elifs = jaxpr_elifs[1:]
+            if pred:
+                return run_jaxpr(jaxpr_elif, *args)
+            return elif_branch(args, rest_preds, rest_jaxpr_elifs)
+
+        def false_branch(args):
+            if jaxpr_false is not None:
+                return run_jaxpr(jaxpr_false, *args)
+            return ()
+
+        if condition:
+            return true_branch(args)
+
+        elif_branch_out = (
+            elif_branch(args, elifs_conditions, jaxpr_elifs) if elifs_conditions.size > 0 else None
+        )
+
+        return false_branch(args) if elif_branch_out is None else elif_branch_out
+
+    @cond_prim.def_abstract_eval
+    def _(*_, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        # We check that the return values in each branch (true, and possibly false and elifs)
+        # have the same abstract values.
+        # The error messages are detailed to help debugging
+        def validate_abstract_values(

```

Comment:
Could this be extracted out of `_get_cond_qfunc_prim`? Looks like it doesn't depend on any closure variables.  

Purely stylistic and personal preference, but I try to reduce nested functions when possible.  Extracting them globally makes them much easier to debug, since they can be imported and run by themselves. It also makes things more readable as you don't have to worry about the namespace and closure variables.